go/hyperkit.go

```diff
@@ -367,32 +375,49 @@ func (h *HyperKit) buildArgs(cmdline string) {
 	a = append(a, "-m", fmt.Sprintf("%dM", h.Memory))
 
 	a = append(a, "-s", "0:0,hostbridge")
+	a = append(a, "-s", "31,lpc")
+
+	nextSlot := 1
+
 	if h.VPNKitSock != "" {
 		if h.VPNKitKey == "" {
-			a = append(a, "-s", fmt.Sprintf("1:0,virtio-vpnkit,path=%s", h.VPNKitSock))
+			a = append(a, "-s", fmt.Sprintf("%d:0,virtio-vpnkit,path=%s", nextSlot, h.VPNKitSock))
 		} else {
-			a = append(a, "-s", fmt.Sprintf("1:0,virtio-vpnkit,path=%s,uuid=%s", h.VPNKitSock, h.VPNKitKey))
+			a = append(a, "-s", fmt.Sprintf("%d:0,virtio-vpnkit,path=%s,uuid=%s", nextSlot, h.VPNKitSock, h.VPNKitKey))
 		}
+		nextSlot++
 	}
+
 	if h.UUID != "" {
 		a = append(a, "-U", h.UUID)
 	}
+
 	if h.DiskImage != "" {
-		a = append(a, "-s", fmt.Sprintf("2:0,virtio-blk,%s", h.DiskImage))
+		a = append(a, "-s", fmt.Sprintf("%d:0,virtio-blk,%s", nextSlot, h.DiskImage))
+		nextSlot++
 	}
+
 	if h.VSock {
-		l := fmt.Sprintf("3,virtio-sock,guest_cid=%d,path=%s", h.VSockGuestCID, h.StateDir)
+		l := fmt.Sprintf("%d,virtio-sock,guest_cid=%d,path=%s", nextSlot, h.VSockGuestCID, h.StateDir)
 		if len(h.VSockPorts) > 0 {
 			l = fmt.Sprintf("%s,guest_forwards=%s", l, intArrayToString(h.VSockPorts, ";"))
 		}
 		a = append(a, "-s", l)
+		nextSlot++
 	}
+
 	if h.ISOImage != "" {
-		a = append(a, "-s", fmt.Sprintf("4,ahci-cd,%s", h.ISOImage))
+		a = append(a, "-s", fmt.Sprintf("%d,ahci-cd,%s", nextSlot, h.ISOImage))
+		nextSlot++
 	}
 
-	a = append(a, "-s", "10,virtio-rnd")
-	a = append(a, "-s", "31,lpc")
```

Comment:
It's not dropped - I moved it up to the other fixed slot, in https://github.com/moby/hyperkit/pull/133/commits/0d990789653c070233cfaff3d788c6b2d26268d8#diff-f91b620a5f864931ab2ac3a6cf135a11R378